docs/fleetautoscaler_spec.md

```diff
@@ -31,12 +49,78 @@ The `spec` field is the actual `FleetAutoscaler` specification and it is compose
                     if not specified, the minimum fleet size will be bufferSize
-    - `maxReplicas` is the maximum fleet size that can be set by this FleetAutoscaler. Required. 
\ No newline at end of file
+    - `maxReplicas` is the maximum fleet size that can be set by this FleetAutoscaler. Required. 
+  - `webhook` parameters of the webhook policy type
+    - `service` parameters
+      - `name`  is the service name bound to Deployment of autoscaler webhook [(see example)](../examples/autoscaler-webhook/autoscaler-service.yaml)
+                      The FleetAutoscaler will scale the fleet up and down based on the response from this webhook server
+      - `namespace` is the kubernetes namespace where webhook is deployed. 
+                      If not specified, the "default" would be used
+      - `path` is the endpoint on the service  (i. e. /autoscale)
+    - `url` is a general URL to make post request to it (optional, instead of service)
+
+Note: only one `buffer` or `webhook` could be defined for FleetAutoscaler depends on `type` field. For Webhook policy we can define either `service` or `url`.
+
+
+# Webhook Endpoint Specification
+
+Webhook endpoint is used to delegate the scaling logic to a separate pod or server.
+
+FleetAutoscaler would send a request to the webhook endpoint every sync period (which is currently 30s) with a JSON body, and scale the target fleet based on the data that is returned.
+JSON payload with a FleetAutoscaleReview data structure would be sent to webhook endpoint and received from it with FleetAutoscaleResponse field populated. FleetAutoscaleResponse contains target Replica count which would trigger scaling of the fleet according to it.
+
+The connection to this webhook endpoint should be defined in `FleetAutoscaler` using Webhook policy type.
+
+```go
+type FleetAutoscaleRequest struct {
+	// UID is an identifier for the individual request/response. It allows us to distinguish instances of requests which are
+	// otherwise identical (parallel requests, requests when earlier requests did not modify etc)
+	// The UID is meant to track the round trip (request/response) between the Autoscaler and the WebHook, not the user request.
+	// It is suitable for correlating log entries between the webhook and apiserver, for either auditing or debugging.
+	UID types.UID `json:"uid""`
+	// Name is the name of the Fleet being scaled
+	Name string `json:"name"`
+	// Namespace is the namespace associated with the request (if any).
+	Namespace string `json:"namespace"`
+	// The Fleet's status values
+	Status v1alpha1.FleetStatus `json:"status"`
+}
+
+type FleetAutoscaleResponse struct {
+	// UID is an identifier for the individual request/response.
+	// This should be copied over from the corresponding FleetAutoscaleRequest.
+	UID types.UID `json:"uid"`
+	// Set to false if no scaling should occur to the Fleet
+	Scale bool `json:"scale"`
+	// The targeted replica count
+	Replicas int32 `json:"replicas"`
```

Comment:
We could have used a pointer value here and avoid the scale part since if it's nil we don't need to scale. WDYT ?

Reply:
Yes, I think it would be easier to read, I followed the spec proposed in the comment from #334